Feat: Accelerators Profile Administration UI e2e Tests

FedeAlonso opened the pull request:
E2e tests for the Accelerator Profile Administration UI (Settings -> Accelerator Profiles):

- Create an Accelerator Profile
- Delete an Accelerator Profile
- Modify an Accelerator Profile
- Disable an Accelerator Profile
- Enable an Accelerator Profile

Test results:
![image](https://github.com/red-hat-data-services/ods-ci/assets/31654558/e1afcb9d-51cd-4371-8292-ea9f485bc4a4)

jstourac commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,154 @@
+    Open ODS Dashboard With Admin User
+    Navigate To Page    Settings    Accelerator profiles
+    Click On Create Accelerator Profile Button
+    Create An Accelerator Profile Via UI   ${ACC_DISPLAY_NAME}01   ${ACC_IDENTIFIER}
```
From the list of the `qe_create_ap_*` files attached to this PR, it looks like the one for this case is missing?

FedeAlonso replied:
This one is going to create the instance using the UI, and those files are used for creating instances using OC  (in order to provision some specific tests). So no need of file for this test.

jstourac commented on ods_ci/tests/Resources/Files/accelerator/qe_create_ap_2.yaml:
```diff
@@ -0,0 +1,33 @@
+kind: AcceleratorProfile
+metadata:
+  generation: 1
+  managedFields:
```
Do we need to have these `managedFields` in all these files?

FedeAlonso replied:
No, we don't really need them. Thanks for pointing it!
Fixed in [ded2638](https://github.com/red-hat-data-services/ods-ci/pull/1209/commits/ded2638dc617dfd07cdd153efcb3d31a485ee7b7)

jstourac commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,154 @@
+Delete Tolerations from an Accelerator Profile Using "Accelerator Profiles" Administration UI
+    [Documentation]    Delete Tolerations from  an Accelerator Profile instance from the Administration UI and
+    ...                verify it's content
+    [Tags]  ODS-XXX
```
What's the plan with all these `ODS-XXX` references?

FedeAlonso replied:
I left that tag waiting to speak with @manosnoam about which one should I use. Fixed in 
a951d55

jstourac commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,154 @@
+Disable an Accelerator Profile From The Accelerator Profile Edit View
+    [Documentation]    Disable an An accelerator profile from the Edit Accelerator Profile view
+    [Tags]  ODS-XXX
+    ...     Smoke
```
I'm thinking... do we want all of these to be part of the `Smoke` testing? :thinking: Maybe keep 2 most important in smoke and move the rest to the `Sanity`?

FedeAlonso replied:
You are right. I've just left the first one as Smoke (as it's the happy path of the feature) and moved everyone else to Sanity. Fixed in [94f171f](https://github.com/red-hat-data-services/ods-ci/pull/1209/commits/94f171f6cf9ddfdcd0c0162592acc8631eb55bea)

jstourac reviewed (commented):
Thank you, Fede.

I looked very briefly and put some comments. Also - just to be sure, this doesn't actually need any GPU to be attached to the cluster, correct?

manosnoam commented:
> E2e tests for the Accelerator Profile Administration UI (Settings -> Accelerator Profiles)

Thank you Fede for these tests!
Can you please add description in more details about the tests, and a screenshot/output of the tests results ?

FedeAlonso commented:
> Thank you, Fede.
> 
> I looked very briefly and put some comments. Also - just to be sure, this doesn't actually need any GPU to be attached to the cluster, correct?

Hi @jstourac , thank you for the review. I've added some fixes :)
No, you don't really need to have a GPU attached. Accelerator Profiles are a CRD that helps the customer to make use of the GPUs, but you can create any instance you want of that resource without engaging that instances to a real GPU

FedeAlonso commented:
> > E2e tests for the Accelerator Profile Administration UI (Settings -> Accelerator Profiles)
> 
> Thank you Fede for these tests! Can you please add description in more details about the tests, and a screenshot/output of the tests results ?

Sure! Thanks for the review @manosnoam 
I've added a brief description of the test added. Also added a description in each of the yaml files.
And the execution results:
![image](https://github.com/red-hat-data-services/ods-ci/assets/31654558/947fc788-9176-4ecd-a0fd-4e0296de5386)

bdattoma commented on ods_ci/tests/Resources/Common.robot:
```diff
@@ -457,3 +457,10 @@ Configure User Workload Monitoring
+Clear element and input text
+    [Documentation]    Clear and input text element, wait .5 seconds and input new text on it
+    [Arguments]    ${element_xpath}    ${new_text}
+            Clear Element Text    ${element_xpath}
+            Sleep    0.5s
+            Input Text    ${element_xpath}    ${new_text}
```
too many tabs

FedeAlonso replied:
Fixed in [a18437b](https://github.com/red-hat-data-services/ods-ci/pull/1209/commits/a18437b54005db759d5041e16facf2aee03b7d00)

bdattoma commented on ods_ci/tests/Resources/Common.robot:
```diff
@@ -457,3 +457,10 @@ Configure User Workload Monitoring
     Log To Console    ${output}
     Should Be Equal As Integers    ${return_code}     0   msg=Error while applying the provided file
+
+Clear element and input text
```
can you use camel case pls?

FedeAlonso replied:
Fixed in [a18437b](https://github.com/red-hat-data-services/ods-ci/pull/1209/commits/a18437b54005db759d5041e16facf2aee03b7d00)

bdattoma commented on ods_ci/tests/Resources/Files/accelerator/qe_create_ap_2.yaml:
```diff
@@ -0,0 +1,15 @@
+  namespace: redhat-ods-applications
+spec:
+  description: description example
+  displayName: qe_create_ap_2
```
could you try with a more realistic displayed name? e.g., QE Accelerator Profile or QE GPU Accelerator etc

FedeAlonso replied:
When a user creates an Accelerator Profile using the UI, when it's asked about the name of that AP, actually the value of that field is the displayName and the name is transformed  eliminating the hyphens and blank spaces. In addition I wanted to maintain certain relation between the name, the displayname and the name of the yaml file used (for that reason I don't use spaces either, so that when doing an "oc create -f xxxx.yaml" it is not necessary to escape the spaces). Besides, keeping this relation between name, displayname and file name, makes it easier to see which resources are used in each test.
Still I agree with you that it is not a good name, so I have changed it to _qeacceleratorprofileX_ and _qe_accelerator_profile_X_.
Fixed in  a18437b

bdattoma commented on ods_ci/tests/Resources/Files/accelerator/qe_create_ap_2.yaml:
```diff
@@ -0,0 +1,15 @@
+apiVersion: dashboard.opendatahub.io/v1
```
why the qe_create_ap_*.yaml start from 2?

bdattoma commented on ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboardSettings.resource:
```diff
@@ -11,6 +11,35 @@ ${GROUP_BTN_XP}=    //button[@data-ouia-component-id="Remove"]
 ${SINGLE_MODE_SERVING_CHECK_BOX}=   //input[@id="single-model-serving-platform-enabled-checkbox"]
 ${MULTI_MODE_SERVING_CHECK_BOX}=   //input[@id="multi-model-serving-platform-enabled-checkbox"]
 ${CUSTOM_EMPTY_GROUP}=    empty-group
+${CREATE_ACCELERATOR_PROFILE_EMPTY_LIST_BTN}=    //button[@data-id="display-accelerator-modal-button"]
+${CREATE_ACCELERATOR_PROFILE_BTN}=    //button[@data-id="create-accelerator-profile"]
+${EDIT_ACCELERATOR_PROFILE_BTN}=    //table[@id='accelerator-profile-table']//span[text()='Edit']/ancestor::button
+${DELETE_ACCELERATOR_PROFILE_BTN}=    //table[@id='accelerator-profile-table']//span[text()='Delete']/ancestor::button
+${ACCELERATOR_PROFILE_GRID_COLUMN_NAME}=    //table[@id='accelerator-profile-table']//td[@data-label='Name']
+${ACCELERATOR_RESOURCES_DIRPATH}=    ods_ci/tests/Resources/Files/accelerator
+${ACCELERATOR_PROFILE_SEARCH_INPUT}=    //input[@aria-label="Search input"]
+${ACCELERATOR_NAME}=    //input[@id="accelerator-name"]
+${ACCELERATOR_IDENTIFIER}=    //input[@id="accelerator-identifier"]
+${ACCELERATOR_DESCRIPTION}=    //textarea[@id="accelerator-description"]
+${ACCELERATOR_ENABLED_SWITCH}=    //input[@id="accelerator-enabled"]/following-sibling::span
+${ACCELERATOR_ADD_TOLERATION_BTN}=    //button[@data-testid="add-toleration-button"]
+${ACCELERATOR_ADD_TOLERATION_MODAL_FORM}=    //div[contains(@id, "pf-modal-part")]//span[text()="Add toleration"]
+${ACCELERATOR_TOLERATION_OPERATOR_DROPDOWN}=    //div[@data-testid="toleration-operator-select"]/button
+${CREATE_OR_UPDATE_ACCELERATOR_PROFILE_DETAIL_BTN}=    //button[@id="create-button"]
+${ACCELERATOR_TOLERATION_EFFECT_DROPDOWN}=    //label[@for="effect-select"]/parent::*/following::*[1]//button
+${ACCELERATOR_TOLERATION_KEY}=    //input[@id='toleration-key']
+${ACCELERATOR_TOLERATION_VALUE}=    //input[@id='toleration-value']
+${ACCELERATOR_TOLERATION_SECONDS_FOREVER_RADIO}=    //input[@id="forever"]
+${ACCELERATOR_TOLERATION_SECONDS_CUSTOM_RADIO}=    //input[@id="custom-value"]
+${ACCELERATOR_TOLERATION_SECONDS_CUSTOM_VALUE}=    //div[contains(@class, 'number-input')]//input
+${ACCELERATOR_TOLERATION_ADD_OR_UPDATE_BUTTON}=    //button[@data-testid="modal-submit-button"]
+${ACCELERATOR_EDIT_NAV_MENU_TOLERATIONS}=    //nav[@aria-label="Jump to section"]//span[text()="Tolerations"]
+${ACCELERATOR_EDIT_TOLERATION_KEBAB_BTN}=    //section[@aria-label="Tolerations"]//table//button[@aria-label="Kebab toggle"]
+${ACCELERATOR_EDIT_TOLERATION_MODAL_FORM}=    //div[contains(@id, "pf-modal-part")]//span[text()="Edit toleration"]
+${ACCELERATOR_DISABLE_MODAL_DISABLE_BUTTON}=    //span[text()='Disable accelerator profile']/ancestor::header/following::footer/button[text()='Disable']
+${ACCELERATOR_DELETE_MODAL_DELETE_BUTTON}=    //span[text()='Delete accelerator profile?']/ancestor::header/following::footer/button[text()='Delete']
+${ACCELERATOR_DELETE_MODAL_INPUT}=    //input[@id="delete-modal-input"]
```
could you pls move the keywords and variables for Accelerator profile in a dedicated resource file like done for ServingRuntimes? `ODHDashboardSettingsRuntimes.resource`

bdattoma commented on ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboardSettings.resource:
```diff
@@ -39,6 +68,16 @@ Add OpenShift Groups To Data Science User Groups
     END
     Press Keys    None    ESC
 
+Open ODS Dashboard With Admin User
+    [Documentation]    Opens a browser and logs into ODS Dashboard with a user belonging to the rhods-admins group
+    Launch Dashboard    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+    ...    ${ODH_DASHBOARD_URL}    browser=${BROWSER.NAME}    browser_options=${BROWSER.OPTIONS}
+
+Open ODS Dashboard With Non Admin User
+    [Documentation]    Opens a browser and logs into ODS Dashboard with a user belonging to the rhods-users group
+    Launch Dashboard    ${TEST_USER_3.USERNAME}    ${TEST_USER_3.PASSWORD}    ${TEST_USER_3.AUTH_TYPE}
+    ...    ${ODH_DASHBOARD_URL}    browser=${BROWSER.NAME}    browser_options=${BROWSER.OPTIONS}
+
```
these don't look like specific for Dashboard settings page. Could you pls move to a more generic common file?

bdattoma commented on ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboardSettings.resource:
```diff
@@ -343,3 +382,267 @@ Set RHODS Users Group To Inexistent Group
+        Click Button   ${CREATE_ACCELERATOR_PROFILE_BTN}
+    END
+
+Click On Edit Accelerator Profile
+    [Documentation]    Click on the the 3 dots button and then on the "Edit" button of an specific Accelerator Profile
+    ...                from the grid
+    ...                view: Settings -> Accelerator profiles
+    [Arguments]   ${name}
+    Search for accelerator profile in grid    ${name}
+    Wait Until Page Contains Element    ${ACCELERATOR_PROFILE_GRID_COLUMN_NAME}//*[text()='${name}']/ancestor::tr//button[@aria-label='Kebab toggle']
+    Click Button    ${ACCELERATOR_PROFILE_GRID_COLUMN_NAME}//*[text()='${name}']/ancestor::tr//button[@aria-label='Kebab toggle']
+    Wait Until Page Contains Element    ${EDIT_ACCELERATOR_PROFILE_BTN}
+    Click Button    ${EDIT_ACCELERATOR_PROFILE_BTN}
```
have you tried re-using `ODHDashboard.Click Action From Actions Menu` keyword? If not, could you pls do it in order to spare new redudant code lines?

bdattoma commented on ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboardSettings.resource:
```diff
@@ -343,3 +382,267 @@ Set RHODS Users Group To Inexistent Group
+    [Documentation]    Disable an specific Accelerator Profile from the Grid
+    ...                view: Settings -> Accelerator profiles
+    [Arguments]   ${name}
+    Click On Enable Accelerator Profile    ${name}
```
the keyword should disable the profile only not enabling it first

bdattoma replied:
it's very ambiguous imo, Could you try renaming it?

FedeAlonso replied:
Done in [8525958](https://github.com/red-hat-data-services/ods-ci/pull/1209/commits/852595828762c1510c8f94885be81a218801e02e)
New keyword: Click On The Enable Switch From Accelerator Profile

bdattoma replied:
I think it's still ambiguous, it's clicking on the Column Header

bdattoma commented on ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboardSettings.resource:
```diff
@@ -343,3 +382,267 @@ Set RHODS Users Group To Inexistent Group
+    Click Element    ${ACCELERATOR_PROFILE_SEARCH_INPUT}
+    Clear element and input text    ${ACCELERATOR_PROFILE_SEARCH_INPUT}   ${name}
+
+Delete Accelerator Profile
+    [Documentation]    Delete an specific Accelerator Profile from the Grid
+    ...                view: Settings -> Accelerator profiles
+    [Arguments]   ${name}
+    Search for accelerator profile in grid    ${name}
+    Wait Until Page Contains Element    ${ACCELERATOR_PROFILE_GRID_COLUMN_NAME}//*[text()='${name}']/ancestor::tr//button[@aria-label='Kebab toggle']
+    Click Button    ${ACCELERATOR_PROFILE_GRID_COLUMN_NAME}//*[text()='${name}']/ancestor::tr//button[@aria-label='Kebab toggle']
+    Wait Until Page Contains Element    ${DELETE_ACCELERATOR_PROFILE_BTN}
+    Click Button    ${DELETE_ACCELERATOR_PROFILE_BTN}
+    Wait Until Page Contains Element    ${ACCELERATOR_DELETE_MODAL_INPUT}
+    Input Text    ${ACCELERATOR_DELETE_MODAL_INPUT}    ${name}
+    Click Button    ${ACCELERATOR_DELETE_MODAL_DELETE_BUTTON}
```
have you tried re-using` ODHDashboard.Click Action From Actions Menu` keyword in combination with `ODHDashboard.Handle Deletion Confirmation Modal?` If not, could you pls do it in order to spare new redudant code lines?

bdattoma commented on ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboardSettings.resource:
```diff
@@ -343,3 +382,267 @@ Set RHODS Users Group To Inexistent Group
+    Input Text    ${ACCELERATOR_DELETE_MODAL_INPUT}    ${name}
+    Click Button    ${ACCELERATOR_DELETE_MODAL_DELETE_BUTTON}
+
+Create An Accelerator Profile Via CLI
+    [Documentation]    Create an instance of Accelerator Profile using OC
+    [Arguments]   ${name}  ${ns}=redhat-ods-applications
+    ${rc}    ${out}    Run And Return Rc And Output   oc create -f ${ACCELERATOR_RESOURCES_DIRPATH}/${name}.yaml -n ${ns}
+    Should Be Equal As Integers    ${rc}     0
```
we are starting splitting CLI resource files from UI ones. Please move it to `ods_ci/tests/Resources/CLI`

bdattoma commented on ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboardSettings.resource:
```diff
@@ -343,3 +382,267 @@ Set RHODS Users Group To Inexistent Group
+    ${rc}    ${out}    Run And Return Rc And Output   oc create -f ${ACCELERATOR_RESOURCES_DIRPATH}/${name}.yaml -n ${ns}
+    Should Be Equal As Integers    ${rc}     0
+
+Create An Accelerator Profile Via UI
```
based on the name of the keyword, I'd assume it clicks the button for creation as first step

bdattoma commented on ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboardSettings.resource:
```diff
@@ -343,3 +382,267 @@ Set RHODS Users Group To Inexistent Group
+    END
+    Click Button    ${CREATE_OR_UPDATE_ACCELERATOR_PROFILE_DETAIL_BTN}
+
+Modify The Accelerator Profile
```
could you call it "Edit Accelerator Profile" for consistency with similar keywords?

bdattoma commented on ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboardSettings.resource:
```diff
@@ -343,3 +382,267 @@ Set RHODS Users Group To Inexistent Group
+    [Arguments]   ${original_display_name}  ${display_name}=${EMPTY}  ${identifier}=${EMPTY}  ${description}=${EMPTY}
+    ...           ${enabled}=${EMPTY}  ${tolerations}=${EMPTY}  ${tol_operator}=${EMPTY}  ${tol_effect}=${EMPTY}
+    ...           ${tol_key}=${EMPTY}  ${tol_value}=${EMPTY}  ${tol_seconds}=${EMPTY}
+    Wait Until Element Is Visible   //h1[text()="Edit ${original_display_name}"]  timeout=10
```
based on the name of the keyword, I'd assume it clicks the button for editing as first step

bdattoma commented on ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboardSettings.resource:
```diff
@@ -343,3 +382,267 @@ Set RHODS Users Group To Inexistent Group
+    IF    "${enabled}" != "${EMPTY}"
+        Click Element    ${ACCELERATOR_ENABLED_SWITCH}
+    END
+    IF    "${tolerations}" != "${EMPTY}"
+        ${first_kebab}=    Catenate    SEPARATOR=    ${ACCELERATOR_EDIT_TOLERATION_KEBAB_BTN}    [1]
+        Click Button    ${first_kebab}
+        Wait Until Element Is Visible    ${first_kebab}/following::div//span[text()="Edit"]    timeout=10
```
this arbitrarly edits the first toleration, is that correct? Isn't there a way to change a specific toleration from the list?

bdattoma commented on ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboardSettings.resource:
```diff
@@ -343,3 +382,267 @@ Set RHODS Users Group To Inexistent Group
+    ${ap_spec_dict}=    Get From Dictionary    ${ap_json}    spec
+    RETURN    ${ap_spec_dict}
+
+Can Not Get Accelerator Profile Via CLI
```
do you mean that the profile should not be found when searched?
Then maybe it would be more clear to call this kw like `Accelerator Profile Should Not Exist` or `Accelerator Profile Should Not Be Found`. It's also a bit more RF style

bdattoma commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,156 @@
+    ...                                    tol_key=${ACC_TOLERATION_KEY}
+    ...                                    tol_value=${ACC_TOLERATION_VALUE}
+    ...                                    tol_seconds=${ACC_TOLERATION_SECONDS}
+    In The Accelerator Profiles Grid There Is An Accelerator Profile With Name   ${ACC_DISPLAY_NAME}01
```
`Accelerator Profile Should Exist` or `Accelerator Profile Should Be Displayed` would be more clear and more consistent with RF style (e.g., Should Be Equal) IMO

bdattoma commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,156 @@
+    In The Accelerator Profiles Grid There Is An Accelerator Profile With Name   ${ACC_DISPLAY_NAME}5
+    Verify Accelerator Profile Values Via CLI   ${ACC_DISPLAY_NAME}5    enabled=True
+
+Disable an Accelerator Profile From The Grid of the Accelerator Profile View
+    [Documentation]    Disable an An accelerator profile from the Edit Accelerator Profile view
+    [Tags]  RHOAIENG-3349
+    ...     Sanity
+    [Setup]  Create An Accelerator Profile Via CLI    ${ACC_DISPLAY_NAME}6
+    Open ODS Dashboard With Admin User
+    Navigate To Page    Settings    Accelerator profiles
+    Search for accelerator profile in grid    ${ACC_DISPLAY_NAME}6
+    Disable Accelerator Profile    ${ACC_NAME}6
+    Verify Accelerator Profile Values Via CLI   ${ACC_DISPLAY_NAME}6    enabled=False
```
you could merge this test case with the previous one to reduce the run duration

FedeAlonso replied:
Around 15 seconds 
![image](https://github.com/red-hat-data-services/ods-ci/assets/31654558/72a46480-0705-4456-ad1a-3d2e9ec29046)
Disabling (or enabling) and verifying takes around 3 seconds.

bdattoma commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,156 @@
+    Open ODS Dashboard With Admin User
+    Navigate To Page    Settings    Accelerator profiles
+    Search for accelerator profile in grid    ${ACC_DISPLAY_NAME}7
+    Click On Enable Accelerator Profile    ${ACC_NAME}7
```
what about adding the check for disabling the profile as well inside this test case?

bdattoma commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,156 @@
+    Navigate To Page    Settings    Accelerator profiles
+    Delete Accelerator Profile    ${ACC_DISPLAY_NAME}8
+    Can Not Get Accelerator Profile Via CLI   ${ACC_NAME}8
+
```
I think we also need a test which runs a workload (e.g., workbench) using a given Accelerator profile to check if it gets all the expected settings defined in the Accelerator profile (e.g, tolerations)

manosnoam replied:
@bdattoma Could that be added on another PR ?

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_create_ap_3.yaml:
```diff
@@ -0,0 +1,15 @@
+  namespace: redhat-ods-applications
+spec:
+  description: description example
+  displayName: qe_create_ap_3
```
Can you change description and displayName with a meaningful name here as well ?

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_create_ap_3.yaml:
```diff
@@ -0,0 +1,15 @@
+  namespace: redhat-ods-applications
+spec:
+  description: description example
+  displayName: qe_create_ap_3
```
Can you change values with a meaningful name here as well:
- name
- description 
- displayName

manosnoam replied:
Hi Fede, the description field is unclear.
The display name also. What is "ap_3" ?
Also, we cannot use underscore in K8s resources values, please use `-` instead.
See https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

FedeAlonso replied:
Also, about the "description" field, as it is not mandatory, maybe it's a good idea to just leave it in the first case (where the user creates an instance manually using the UI).
WDYT @manosnoam ?

manosnoam replied:
I would make the description and displayname more relevant to the test used with that yaml.
For example, you can set displayName = "Testing accelerator with display name contains special characters". As well as the description - it should be relevant for the actual test. 

A good option would be to use `${TEST NAME}`and `${TEST DOCUMENTATION}` ([builtin variables in RF](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#automatic-variables)) as the description field value.
You can do it by replacing the value with [Jinja template mechanism](https://codeburst.io/jinja-2-explained-in-5-minutes-88548486834e) which we already support with our [openshift robot library](https://github.com/red-hat-data-services/robotframework-openshiftcli), for example:
`Oc Apply  AcceleratorProfile  qe_accelerator_profile_3.yaml  template_data={description: '${TEST DOCUMENTATION}', displayName: '${TEST NAME}'}`

FedeAlonso replied:
Also, I'm using DisplayName as a useful variable to manage all tests with only adding a number at the end:
`${ACC_DISPLAY_NAME}=    qe_accelerator_profile_X`
I'm doing it so then I can filter by that value easily

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_create_ap_5.yaml:
```diff
@@ -0,0 +1,11 @@
+  namespace: redhat-ods-applications
+spec:
+  description: description example
+  displayName: qe_create_ap_5
```
Can you change values with a meaningful name here as well:
- name
- description 
- displayName

manosnoam replied:
Hi Fede, the name, description and displayNames fields are yet unclear.
What are "example" and "ap_" ?

Also, we should not use underscore in K8s resources values, please use `-` instead.
See https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
The display name could be as the description I believe, but in the name please split words with `-`.

FedeAlonso replied:
Same as above

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_create_ap_6.yaml:
```diff
@@ -0,0 +1,16 @@
+  namespace: redhat-ods-applications
+spec:
+  description: description example
+  displayName: qe_create_ap_6
```
Can you change values with a meaningful name here as well:
- name
- description 
- displayName

manosnoam replied:
Hi Fede, the name, description and displayNames fields are yet unclear.
What are "example" and "ap_" ?

Also, we should not use underscore in K8s resources values, please use `-` instead.
See https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
The display name could be as the description I believe, but in the name please split words with `-`.

FedeAlonso replied:
Same as above

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_create_ap_7.yaml:
```diff
@@ -0,0 +1,16 @@
+  namespace: redhat-ods-applications
+spec:
+  description: description example
+  displayName: qe_create_ap_7
```
Can you change values with a meaningful name here as well:
- name
- description 
- displayName

manosnoam replied:
Hi Fede, the name, description and displayNames fields are yet unclear.
What are "example" and "ap_" ?

Also, we should not use underscore in K8s resources values, please use `-` instead.
See https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
The display name could be as the description I believe, but in the name please split words with `-`.

FedeAlonso replied:
Same as above

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_create_ap_8.yaml:
```diff
@@ -0,0 +1,11 @@
+  namespace: redhat-ods-applications
+spec:
+  description: description example
+  displayName: qe_create_ap_8
```
Can you change values with a meaningful name here as well:
- name
- description 
- displayName

manosnoam replied:
Hi Fede, the name, description and displayNames fields are yet unclear.
What are "example" and "ap_" ?

Also, we should not use underscore in K8s resources values, please use `-` instead.
See https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
The display name could be as the description I believe, but in the name please split words with `-`.

FedeAlonso commented:
After all this changes, I've ran  the tests agains

![image](https://github.com/red-hat-data-services/ods-ci/assets/31654558/57e17495-91ac-4f30-85e7-3c7faa8bcf73)

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_create_ap_4.yaml:
```diff
@@ -0,0 +1,16 @@
+  namespace: redhat-ods-applications
+spec:
+  description: description example
+  displayName: qe_create_ap_4
```
Hi Fede, the description field is unclear.
The display name also. What is "ap_4" ?
Also, we cannot use undescore in K8s resources values, please use `-` instead.
See https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

FedeAlonso replied:
Resolving this comment in order to just have one thread: https://github.com/red-hat-data-services/ods-ci/pull/1209/files/8799dfb660640bd8d4ce1f95cfa357dd9e18b521#r1494746578
I will apply the conclussions to all cases

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_accelerator_profile_2.yaml:
```diff
@@ -0,0 +1,17 @@
+apiVersion: dashboard.opendatahub.io/v1
+kind: AcceleratorProfile
+metadata:
+  name: qeacceleratorprofile2
+  namespace: redhat-ods-applications
+spec:
+  description: description example
+  displayName: qe_accelerator_profile_2
```
```suggestion
  name: accelerator-profile-test-2
  namespace: redhat-ods-applications
spec:
  description: You can give a meaningful description here, but if it is obvious, just remove this field completely.
  displayName: Accelerator Profile Test 2
```

FedeAlonso replied:
Not resolving it yet. Waitint for a reply above.

manosnoam reviewed (changes requested):
Hi Fede, I added comments to rename fields values in the yaml files, with one example

bdattoma commented on ods_ci/tests/Resources/Common.robot:
```diff
@@ -457,3 +457,10 @@ Configure User Workload Monitoring
+Clear Element And Input Text
+    [Documentation]    Clear and input text element, wait .5 seconds and input new text on it
+    [Arguments]    ${element_xpath}    ${new_text}
+        Clear Element Text    ${element_xpath}
```
too many tabs

FedeAlonso replied:
Fixed in 
1e6d6e0

bdattoma commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,153 @@
+    [Tags]  RHOAIENG-3349
+    ...     Sanity
+    [Setup]  Create An Accelerator Profile Via CLI    ${ACC_DISPLAY_NAME}3
+    Open ODS Dashboard With Admin User
+    Navigate To Page    Settings    Accelerator profiles
```
these 2 lines are repated in every test. Maybe could become a test setup

FedeAlonso replied:
Good point.
Done in 
1e6d6e0

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_accelerator_profile_6.yaml:
```diff
@@ -0,0 +1,17 @@
+  namespace: redhat-ods-applications
+spec:
+  displayName: qe_accelerator_profile_6
+  enabled: true
```
Shouldn't it be "false" (disable) ?

FedeAlonso replied:
As in the test I'm disabling the AP, I need to provision one enabled.

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_accelerator_profile_6.yaml:
```diff
@@ -0,0 +1,17 @@
+  name: qeacceleratorprofile6
+  namespace: redhat-ods-applications
+spec:
+  displayName: qe_accelerator_profile_6
```
I would set `displayName` and `description` according to the comments in the first 2 lines (or as I suggested with ${TEST NAME} and ${TEST DOCUMENTATION})

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_accelerator_profile_7.yaml:
```diff
@@ -0,0 +1,17 @@
+  namespace: redhat-ods-applications
+spec:
+  displayName: qe_accelerator_profile_7
+  enabled: false
```
Shouldn't it be "true" (enable) ?

FedeAlonso replied:
In the test I'm enabling the AP, I need to provision one disabled.

manosnoam commented on ods_ci/tests/Resources/CLI/AcceleratorProfiles/AcceleratorProfiles.resource:
```diff
@@ -0,0 +1,96 @@
+Create An Accelerator Profile Via CLI
+    [Documentation]    Create an instance of Accelerator Profile using OC
+    [Arguments]   ${name}  ${ns}=redhat-ods-applications
+    ${rc}    ${out}    Run And Return Rc And Output   oc create -f ${ACCELERATOR_RESOURCES_DIRPATH}/${name}.yaml -n ${ns}
```
@FedeAlonso Can you fix ` The assignment sign is not consistent within the file. Expected '{{ expected_sign }}' but got '{{ actual_sign }}' instead ` 
You probably need `${rc}    ${out}=` ...

FedeAlonso replied:
Fixed in 
822a36b

manosnoam commented on ods_ci/tests/Resources/CLI/AcceleratorProfiles/AcceleratorProfiles.resource:
```diff
@@ -0,0 +1,96 @@
+    ${rc}    ${out}    Run And Return Rc And Output   oc create -f ${ACCELERATOR_RESOURCES_DIRPATH}/${name}.yaml -n ${ns}
+    Should Be Equal As Integers    ${rc}     0
+
+Get Accelerator Profile Spec Values via CLI
```
Can you fix Camel Case Keywords please.

FedeAlonso replied:
Fixed in 
822a36b

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_accelerator_profile_2.yaml:
```diff
@@ -0,0 +1,17 @@
+apiVersion: dashboard.opendatahub.io/v1
+kind: AcceleratorProfile
+metadata:
+  name: qeacceleratorprofile2
```
Just an example, but relevant for all yamls:
```suggestion
  name: accelerator-profile-2
```

FedeAlonso replied:
I do not think It's a good idea to delete the "qe". The idea here is to have some kind of string which is unique for this feature. Something like  "qe-accelerator-profile-XXX" so when this feature (or test suite) has ended, we can filter all which start with "qe-accelerator-profile-" and delete it (as we are already doing). If we delete the "qe" and run this is some productuion cluster and there are AP instances which start with name "accelerator-profile" we will delete them

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_accelerator_profile_2.yaml:
```diff
@@ -0,0 +1,17 @@
+  name: qeacceleratorprofile2
+  namespace: redhat-ods-applications
+spec:
+  description: Modify Accelerator Profile
```
Just an example, but relevant for all yamls:
```suggestion
  description: Test: Modify An Accelerator Profile Using "Accelerator Profiles" Administration UI
```

manosnoam commented on ods_ci/tests/Resources/Files/accelerator/qe_accelerator_profile_2.yaml:
```diff
@@ -0,0 +1,17 @@
+  namespace: redhat-ods-applications
+spec:
+  description: Modify Accelerator Profile
+  displayName: qe_accelerator_profile_2
```
Just an example, but relevant for all yamls:
```suggestion
  displayName: Test: Modify Accelerator Profile 2
```
This is more clear than with underscores + you test `:` which is a special character (even more than `_`)

manosnoam reviewed (changes requested):
I mostly added comments for `qe_accelerator_profile_2.yaml`, but they are relevant for all other yamls as well, if you choose to accept ;-)

FedeAlonso commented:
> I mostly added comments for `qe_accelerator_profile_2.yaml`, but they are relevant for all other yamls as well, if you choose to accept ;-)

I've accepted most of them, and also I've used Jinja

FedeAlonso commented:
![image](https://github.com/red-hat-data-services/ods-ci/assets/31654558/e5543a54-68a3-4c86-bf19-a9f7d70da7bb)
Latest test execution

jstourac commented on ods_ci/tests/Resources/CLI/AcceleratorProfiles/AcceleratorProfiles.resource:
```diff
@@ -0,0 +1,98 @@
+Get Accelerator Profile Spec Values Via CLI
+    [Documentation]  Get Spec fields from an specific Accelerator Profile Instance
+    [Arguments]   ${display_name}    ${ns}=redhat-ods-applications
+    ${ap_from_oc}=    Run     oc get acceleratorprofile -n ${ns} -o json | jq '.items[] | select(.spec.displayName == "${display_name}")'
```
Cheap robocop warning reduction

```suggestion
    ${ap_from_oc}=    Run
    ...    oc get acceleratorprofile -n ${ns} -o json | jq '.items[] | select(.spec.displayName == "${display_name}")'
```

jstourac commented on ods_ci/tests/Resources/CLI/AcceleratorProfiles/AcceleratorProfiles.resource:
```diff
@@ -0,0 +1,98 @@
+    ...                value of the OC Accelerator Profile resource
+    ...                Uses OC
+    [Arguments]   ${ap_prefix}  ${ns}=redhat-ods-applications
+    ${ap_str}=    Run     oc get acceleratorprofiles -n ${ns} -o json | jq .items[].metadata.name | grep ${ap_prefix} | tr -d '"'
```
Cheap robocop warning reduction

```suggestion
    ${ap_str}=    Run
    ...    oc get acceleratorprofiles -n ${ns} -o json | jq .items[].metadata.name | grep ${ap_prefix} | tr -d '"'
```

jstourac commented on ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboard.resource:
```diff
@@ -1,5 +1,6 @@
 *** Settings ***
 Resource   ODHDashboard.robot
+Resource        ../../LoginPage.robot
```
indentation?

FedeAlonso replied:
Fixed in [beb40ed](https://github.com/red-hat-data-services/ods-ci/pull/1209/commits/beb40ed07323d2aa0b501317592df34828c74619)

jstourac commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,167 @@
+    [Documentation]    Create an Accelerator Profile instance from the Administration UI and verify it's content
+    [Tags]  RHOAIENG-3349
+    ...     Smoke
+    [Setup]  Open Dashboard And Navigate to Accelerator Profiles From Settings
```
Also eventual cheap robocop warning fix

```suggestion
    [Setup]  Open Dashboard And Navigate To Accelerator Profiles From Settings
```

jstourac reviewed (approved):
Hi Fede, I had a very brief look again. In general looks good to me.

I wrote some nitpick comments on a few places. This branch needs to be rebased.

Also, pleas don't forget to squash the commits before the merge!!! :slightly_smiling_face:

FedeAlonso commented:
![image](https://github.com/red-hat-data-services/ods-ci/assets/31654558/5167f6ec-6136-4348-bd1c-da459aa1173d)
Again, with the latest changes

jgarciao commented on ods_ci/tests/Resources/CLI/AcceleratorProfiles/AcceleratorProfiles.resource:
```diff
@@ -0,0 +1,100 @@
+*** Keywords ***
+Create An Accelerator Profile Via CLI
+    [Documentation]    Create an instance of Accelerator Profile using OC
+    [Arguments]   ${name}  ${ns}=redhat-ods-applications
```
Instead of redhat-ods-applications it should be `${APPLICATIONS_NAMESPACE}` to make it compatible with odh-nightly. Same comment for other places in the same PR

FedeAlonso replied:
Done [45138df](https://github.com/red-hat-data-services/ods-ci/pull/1209/commits/45138df54bbb0bd0091701d89f9da46341d8b3c6)****

jgarciao commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,167 @@
+Create An Accelerator Profile From Accelerator Profiles Administration UI
+    [Documentation]    Create an Accelerator Profile instance from the Administration UI and verify it's content
+    [Tags]  RHOAIENG-3349
+    ...     Smoke
```
Normally we add the tests to Smoke when they have been tested for a while and they are proven to be stable. Do you mind adding it to Sanity and Tier1 now?

FedeAlonso replied:
Done [110b73a](https://github.com/red-hat-data-services/ods-ci/pull/1209/commits/110b73a6aa09d7ed1e07c3691624829a284a5a3b)

jgarciao commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,167 @@
+Modify An Accelerator Profile Using Accelerator Profiles Administration UI
+    [Documentation]    Modify an Accelerator Profile instance from the Administration UI and verify it's content
+    [Tags]  RHOAIENG-3349
+    ...     Sanity
```
It should be also in Tier1

jgarciao commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,167 @@
+    [Documentation]    Delete Tolerations from  an Accelerator Profile instance from the Administration UI and
+    ...                verify it's content
+    [Tags]  RHOAIENG-3349
+    ...     Sanity
```
It should be also in Tier1

FedeAlonso replied:
Done [110b73a](https://github.com/red-hat-data-services/ods-ci/pull/1209/commits/110b73a6aa09d7ed1e07c3691624829a284a5a3b)

jgarciao commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,167 @@
+Disable an Accelerator Profile From The Accelerator Profile Edit View
+    [Documentation]    Disable an An accelerator profile from the Edit Accelerator Profile view
+    [Tags]  RHOAIENG-3349
+    ...     Sanity
```
It should be also in Tier1

jgarciao commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,167 @@
+    Accelerator Profile Should Be Displayed In The Grid   ${ACC4_DISPLAY_NAME}
+    Verify Accelerator Profile Values Via CLI   ${ACC4_DISPLAY_NAME}    enabled=False
+
+Enable an Accelerator Profile From The Accelerator Profile Edit View
```
It should be also in Tier1

FedeAlonso replied:
Done [110b73a](https://github.com/red-hat-data-services/ods-ci/pull/1209/commits/110b73a6aa09d7ed1e07c3691624829a284a5a3b)

jgarciao commented on ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot:
```diff
@@ -0,0 +1,167 @@
+Teardown Settings Accelerator Profiles
+    [Documentation]    Sets the default values In User Management Settings
+    ...                and runs the RHOSi Teardown
+    Dashboard Settings Accelerator Profiles Test Teardown
```
Shouldn't the test teardown be called when finishing each test too?

FedeAlonso replied:
In the teardown the test just delete the created resources (Accelerator profile instances in this case).
The way I've done it is using the same preffix for all the instances, so it witll do an "oc get accelratorprofiles ..." filter the ones with the preffix and  then delete 'em.
I think that it's better to do this way instead of deleting each resource after each test because:
1. It's centralized
2. If someone wants to add some new tests, and follow the name structure as in the older tests (which is easy to realize because all tests use the variable ACC_NAME), does not need to delete anything

bdattoma replied:
probably you don't need the `Dashboard Settings Accelerator Profiles Test Teardown` keyword, but you could just move `Delete All Accelerator Profiles Which Starts With ${ACC_NAME}` inside the Suite Teardown

FedeAlonso replied:
Done [ecbabe7](https://github.com/red-hat-data-services/ods-ci/pull/1209/commits/ecbabe7a2c2442c9ab91cc5cd637dc11eac4a05d)

FedeAlonso commented:
Execution with latest changes: 
![image](https://github.com/red-hat-data-services/ods-ci/assets/31654558/b08df8fd-89c4-40d2-bfd7-40a1235506a4)

FedeAlonso commented:
![image](https://github.com/red-hat-data-services/ods-ci/assets/31654558/334e10b8-c8c4-4da7-8cba-62bc36c0921a)

manosnoam reviewed (approved):
Thanks Fede for adding all of these tests!

jgarciao commented on ods_ci/tests/Resources/Files/accelerator/accelerator-profile-3349-2.yaml:
```diff
@@ -0,0 +1,17 @@
+kind: AcceleratorProfile
+metadata:
+  name: accelerator-profile-3349-2
+  namespace: redhat-ods-applications
```
Can you set `namespace` as a parameter (same as you do on the `description` field) so it can be set to
`${APPLICATIONS_NAMESPACE}` ? Same for the other yaml files

FedeAlonso replied:
Sure! Done [f8c06b4](https://github.com/red-hat-data-services/ods-ci/pull/1209/commits/f8c06b4dbff37e37839d68cacc141f21cbf84446)

FedeAlonso commented:
Test results with the latests changes: 
![image](https://github.com/red-hat-data-services/ods-ci/assets/31654558/7b482d85-2acd-407c-839a-d0b829397681)
